library/alloc/src/vec/mod.rs

```diff
@@ -1023,8 +1023,10 @@ impl<T, A: Allocator> Vec<T, A> {
 
     /// Shrinks the capacity of the vector as much as possible.
     ///
-    /// It will drop down as close as possible to the length but the allocator
-    /// may still inform the vector that there is space for a few more elements.
+    /// The behavior of this method depends on the allocator being used. If the allocator supports
+    /// in-place shrinking, some or all of the excess capacity may be returned to the allocator.
+    /// If not, this method may cause the vector's data to be copied to a new allocation with
+    /// exactly the correct size. See [`Allocator::shrink`] for more details.
```

Comment:
We don't guarantee "exactly" though. The same fuzziness as `with_capacity` or `reserve_exact` applies.

Reply:
Got it, this claim was made in at least one other place that I found and fixed.